conans/test/functional/toolchains/meson/test_ios.py

```diff
@@ -56,33 +59,17 @@ def build(self):
     def settings(self):
         return [("os", self.os),
                 ("os.version", self.os_version),
+                ("os.sdk", self.os_sdk),
                 ("arch", self.arch),
                 ("compiler", "apple-clang"),
                 ("compiler.version", "12.0"),
                 ("compiler.libcxx", "libc++")]
 
-    def env(self):
-        cc = self.xcrun.cc
-        cxx = self.xcrun.cxx
-
-        deployment_flag = apple_deployment_target_flag(self.os, self.os_version)
```

Comment:
Users shouldn't be doing this in their profile code, this is exactly the purpose of ``MesonToolchain``, lets move this logic there.

Reply:
see #9710 (and #9711), it's opened exactly for that.
in general, I am not a fan of doing several unrelated things in one PR. let's continue the discussion on how to do blocks in the dedicated PR.